cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/TooManyRequestsException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.provisioner;
+
+/**
+ * Thrown when throttling occurs while starting a job.
+ */
+public class TooManyRequestsException extends Exception {
```

Comment:
There is a TooManyRequestsException in cdap-common. Do you intend to keep both?

Reply:
I don't think dataproc runtime extension should have a dependency on cdap-common. The only cdap dependency it has presently is cdap-runtime-spi which provides the interfaces and classes to be extended. This is the reason I decided to duplicate the exception in runtime-spi. Is there a better way?